### PR TITLE
Prevent from buying scopes and ballistic module in HvH

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Title. Can't remove these items from all vendor

## Changelog
:cl:
balance: Prevent from buying scopes and ballistic module in HvH
/:cl:

### CODE REVIEW

**Comment by SplinterGP** [comment]
i think maybe scopes could be limited on HvH?

**Comment by BraveMole** [comment]
> i think maybe scopes could be limited on HvH?

Snipers in req

**Comment by SplinterGP** [comment]
coulnd't you remove them on HVH vendor and just spawn these on the twin pillars instead ?

File: code/_globalvars/lists/typechache.dm [hunk 1]
```diff
@@ -8,3 +8,8 @@ GLOBAL_LIST_INIT(typecache_mob, typecacheof(/mob))
 GLOBAL_LIST_INIT(typecache_living, typecacheof(/mob/living))
 
 GLOBAL_LIST_INIT(ignored_atoms, typecacheof(list(/mob/dead, /obj/effect/landmark, /obj/docking_port, /obj/effect/particle_effect/sparks, /obj/effect/DPtarget, /obj/effect/supplypod_selector, /atom/movable/lighting_object)))
+
+GLOBAL_LIST_INIT(hvh_restricted_items_list, typecacheof(list(
+	/obj/item/armor_module/attachable/ballistic_armor,
+	/obj/item/attachable/scope,
+)))
```
Comment (BraveMole) [inline code/_globalvars/lists/typechache.dm:15]:
```suggestion
)))

```

File: code/game/objects/machinery/vending/vending.dm [hunk 2]
```diff
@@ -563,7 +563,12 @@
 /obj/machinery/vending/proc/vend(datum/vending_product/R, mob/user)
 	if(!allowed(user) && (!wires.is_cut(WIRE_IDSCAN) || hacking_safety)) //For SECURE VENDING MACHINES YEAH
 		to_chat(user, "<span class='warning'>Access denied.</span>")
-		flick(src.icon_deny,src)
+		flick(icon_deny, src)
+		return
+	
+	if(SSticker.mode?.flags_round_type & MODE_HUMAN_ONLY && is_type_in_typecache(R.product_path, GLOB.hvh_restricted_items_list))
+		to_chat(user, "<span class='warning'>This item was banned by the Space Geneva Convention.</span>")
```
Comment (TiviPlus) [inline code/game/objects/machinery/vending/vending.dm:570]:
```suggestion
		to_chat(user, "<span class='warning'>This item is banned by the Space Geneva Convention.</span>")
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**